src/rgw/CMakeLists.txt

```diff
@@ -249,38 +249,39 @@ endif()
 
 target_link_libraries(rgw_common
   PRIVATE
-    global
-    cls_2pc_queue_client
-    cls_cmpomap_client
-    cls_lock_client
-    cls_log_client
-    cls_otp_client
-    cls_refcount_client
-    cls_rgw_client
-    cls_rgw_gc_client
-    cls_timeindex_client
-    cls_user_client
-    cls_version_client
-    librados
-    rt
-    ICU::uc
-    OATH::OATH
-    dmclock::dmclock
-    ${CURL_LIBRARIES}
-    ${EXPAT_LIBRARIES}
-    ${ARROW_LIBRARIES}
-    ${ARROW_FLIGHT_LIBRARIES}
-    ${ALLOC_LIBS}
+  global
+  cls_2pc_queue_client
+  cls_cmpomap_client
+  cls_lock_client
+  cls_log_client
+  cls_otp_client
+  cls_refcount_client
+  cls_rgw_client
+  cls_rgw_gc_client
+  cls_timeindex_client
+  cls_user_client
+  cls_version_client
+  librados
+  $<$<AND:$<BOOL:UNIX>,$<NOT:$<BOOL:APPLE>>>:rt>
+  ICU::uc
+  OATH::OATH
+  dmclock::dmclock
+  ${CURL_LIBRARIES}
+  ${EXPAT_LIBRARIES}
+  ${ARROW_LIBRARIES}
+  ${ARROW_FLIGHT_LIBRARIES}
+  ${ALLOC_LIBS}
   PUBLIC
-    ${LUA_LIBRARIES}
-    RapidJSON::RapidJSON
-    spawn
-    fmt::fmt)
+  ${LUA_LIBRARIES}
+  RapidJSON::RapidJSON
+  spawn
+  fmt::fmt)
 target_include_directories(rgw_common
   PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw/services"
   PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw/driver/rados"
   PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw"
-  PUBLIC "${LUA_INCLUDE_DIR}")
+  PUBLIC "${LUA_INCLUDE_DIR}"
+  PUBLIC "${OPENSSL_INCLUDE_DIR}")
```

Comment:
👍 i remember that this was necessary to build with `WITH_RADOSGW_AMQP_ENDPOINT=OFF` because of this:
```cmake
if(WITH_RADOSGW_AMQP_ENDPOINT)
  # used by rgw_amqp.cc
  target_link_libraries(rgw_common
    PRIVATE
      RabbitMQ::RabbitMQ
      OpenSSL::SSL)
endif()
```
but instead of adding just the `OPENSSL_INCLUDE_DIR` here, please add the `OpenSSL::SSL` target to the `PUBLIC` section of `target_link_libraries(rgw_common)`, and remove it from this `WITH_RADOSGW_AMQP_ENDPOINT` case and the other dependent targets that add it